src/Domain/Structure/ElementSearchTree.hpp

```diff
@@ -96,9 +39,44 @@ namespace domain {
  * choices work well, but haven't been extensively tuned.
```

Comment:
Could you add to the documentation why we have a wrapper with PIMPL? Just so we don't undo it in the future :)